components/homme/src/share/reduction_mod.F90

```diff
@@ -337,6 +342,92 @@ subroutine pmax_mt_r_1d(red,redp,len,hybrid)
   end subroutine pmax_mt_r_1d
 
 
+!max with location index
+!at the end result is in redp thread local array
+  subroutine ParallelMaxWithIndex(redp,hybrid)
+    use hybrid_mod, only : hybrid_t
+#ifdef _MPI
+    use parallel_mod, only: mpi_maxloc, mpi2real_t
+#endif
+
+    real (kind=real_kind), intent(inout) :: redp(2) ! thread private vector of partial sum
+    type (hybrid_t),       intent(in)    :: hybrid  ! parallel handle
+
+    ! Local variables
+    integer ierr, k
+
+    ! the first and fastest thread performs initializing copy
+    !$OMP SINGLE
```

Comment:
I forgot: are there supposed to be "#ifdef HORIZ_OPENMP" around all the OMP lines?

Reply:
I used pmax_mt_r_1d() as an example for these routines, and there is no HORIZ_OPENMP. But it seems that for column threads all these lines of code would be active (with duplicated buffers?), so, is it a mistake to not to wrap everything in reduction mod in ifdef HORIZ_OPENMP? Though I think there will be no column threads in homme in future...